packages/js/data/src/reports/selectors.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Internal dependencies
+ */
+import { getResourceName } from '../utils';
+import {
+	ReportState,
+	ReportItemsEndpoint,
+	ReportStatEndpoint,
+	ReportQueryParams,
+	ReportStatQueryParams,
+	ReportItemObjectInfer,
+	ReportStatObjectInfer,
+} from './types';
+
+const EMPTY_OBJECT = {} as const;
+
+export const getReportItemsError = (
+	state: ReportState,
+	endpoint: ReportItemsEndpoint,
+	query: ReportQueryParams
+) => {
+	const resourceName = getResourceName( endpoint, query );
+	return state.itemErrors[ resourceName ] || false;
+};
+
+export const getReportItems = < T >(
+	state: ReportState,
+	endpoint: ReportItemsEndpoint,
+	query: ReportQueryParams
+): ReportItemObjectInfer< T > => {
+	const resourceName = getResourceName( endpoint, query );
+	return (
+		( state.items[ resourceName ] as ReportItemObjectInfer< T > ) ||
```

Comment:
if getResourceName is typed strongly I think we should be able to do without the type coercion here since we can index state.items if resourceName is typed

Reply:
We need to do type coercion here because the item value type is `ReportStatObject` but we want to return `ReportItemObjectInfer< T >`.

We may need something like below but  I think Typescript doesn't support that. (?)

```ts
export type ReportState = {
         // ....
	stats: {
		[`products:*`]: ProductReportStat;
	};
};
```